### ISSUE TITLE
`make format` target that formats all code

### ISSUE THREAD

**Issue by andybalaam** [andybalaam/rabbit-escape#357]
It is possible to run Eclipse's code formatter from the command line*.

> \* although I think only in a graphical environment

Eclipse's formatter is surprisingly effective and comprehensive - better than IDEA's (for my preferred style anyway) and better than anything else I've seen.  If anyone knows better, please let me know because I'd much rather avoid depending on Eclipse.

So, make a Make target that formats all the Java code in the project (including Android code) in our preferred style.

Also, write instructions for IDEA and Android Studio to allow them to use the [Eclipse Code Formatter](http://plugins.jetbrains.com/plugin/?idea&id=6546) plugin to share the same coding style.

**Comment by colonelfazackerley** [comment]
Sometimes it's nice to put some extra spaces in to get something to line up.

I guess we need to decide if it's worth trading this in for more uniformity.

**Comment by colonelfazackerley** [comment]
Aha. Tags can be used to turn off the formatter in certain regions.

http://stackoverflow.com/questions/5115088/turn-off-eclipse-formatter-for-selected-code-area

**Comment by andybalaam** [comment]
Yes, that's the nuclear option, and we can also adjust the format to allow things we like.

**Comment by andybalaam** [comment]
[Artistic Style](http://astyle.sourceforge.net/astyle.html) looks like it might do what we need.

**Comment by andybalaam** [comment]
I am considering starting a small project called Kevlin that runs a code formatter that formats code as Kevlin would have it.
